codegen/render/manifests_renderer.go

```diff
@@ -26,60 +26,84 @@ import (
-type MakeResourceFunc func(group Group) ([]metav1.Object, error)
+type MakeResourceFunc func(groups []*Group) ([]metav1.Object, error)
 
 // renders kubernetes from templates
 type ManifestsRenderer struct {
+	templateRenderer
 	AppName       string // used for labeling
 	ResourceFuncs map[OutFile]MakeResourceFunc
 	ManifestDir   string
 	ProtoDir      string
 }
 
+type templateArgs struct {
+	Crds       []apiextv1.CustomResourceDefinition
+	ShouldSkip map[string]bool
+}
+
 func RenderManifests(
 	appName, manifestDir, protoDir string,
 	protoOpts protoutil.Options,
 	groupOptions model.GroupOptions,
-	grp Group,
+	grps []*Group,
 ) ([]OutFile, error) {
 	defaultManifestsRenderer := ManifestsRenderer{
 		AppName:     appName,
 		ManifestDir: manifestDir,
 		ProtoDir:    protoDir,
-		ResourceFuncs: map[OutFile]MakeResourceFunc{
-			{
-				Path: manifestDir + "/crds/" + grp.Group + "_" + grp.Version + "_" + "crds.yaml",
-			}: func(group Group) ([]metav1.Object, error) {
-				return kuberesource.CustomResourceDefinitions(group)
-			},
-		},
-	}
-	return defaultManifestsRenderer.RenderManifests(grp, protoOpts, groupOptions)
+	}
+	return defaultManifestsRenderer.RenderManifests(grps, protoOpts, groupOptions)
 }
 
-func (r ManifestsRenderer) RenderManifests(grp Group, protoOpts protoutil.Options, groupOptions model.GroupOptions) ([]OutFile, error) {
-	if !grp.RenderManifests {
-		return nil, nil
+func (r ManifestsRenderer) RenderManifests(grps []*Group, protoOpts protoutil.Options, groupOptions model.GroupOptions) ([]OutFile, error) {
+	grpsByGroupName := make(map[string][]*Group)
+	shouldRenderGroups := make(map[string]bool)
+	grandfatheredGroups := make(map[string]bool)
+	for _, grp := range grps {
+		grpsByGroupName[grp.Group] = append(grpsByGroupName[grp.Group], grp)
+		shouldRenderGroups[grp.Group] = shouldRenderGroups[grp.Group] || grp.RenderManifests
+		if grp.Grandfathered {
+			grandfatheredGroups[grp.GroupVersion.String()] = true
+		}
 	}
 
-	if grp.RenderValidationSchemas {
-		var err error
-		oapiSchemas, err := generateOpenApi(grp, r.ProtoDir, protoOpts, groupOptions)
-		if err != nil {
-			return nil, err
+	for _, grp := range grps {
+		if grp.RenderValidationSchemas && shouldRenderGroups[grp.Group] {
+			var err error
+			oapiSchemas, err := generateOpenApi(*grp, r.ProtoDir, protoOpts, groupOptions)
+			if err != nil {
+				return nil, err
+			}
+			grp.OpenApiSchemas = oapiSchemas
 		}
-		grp.OpenApiSchemas = oapiSchemas
 	}
 
 	var renderedFiles []OutFile
-	for out, mkFunc := range r.ResourceFuncs {
-		content, err := r.renderManifest(r.AppName, mkFunc, grp)
+
+	for groupName, selectedGrps := range grpsByGroupName {
+		if !shouldRenderGroups[groupName] {
+			continue
+		}
+
+		crds, err := r.createCrds(r.AppName, selectedGrps)
+		if err != nil {
+			return nil, err
+		}
+		// TODO (dmitri-d): this can be removed once we migrate to use platform charts exclusively
+		out, err := r.renderLegacyManifest(r.AppName, groupName, crds)
```

Comment:
We would leave this since Edge uses it as well as some extensions, maybe let's not use the word legacy, since generation to the crd folder is just a different use-case. I would add conditionals for each so they can be selectively enabled. I would call this renderCRDManfiest & renderTemplatedCRDManifest. The vars to enabled them can be similar.

Reply:
done